Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocViewController.swift

```diff
@@ -0,0 +1,374 @@
+                return estimate
+            }
+            
+            
+            largeEventCell.configure(with: largeEvent, theme: theme)
+            cell = largeEventCell
+        case .small(let smallEvent):
+            guard let smallEventCell = layoutManager.placeholder(forCellWithReuseIdentifier: ArticleAsLivingDocSmallEventCollectionViewCell.identifier) as? ArticleAsLivingDocSmallEventCollectionViewCell else {
+                return estimate
+            }
+            
+            smallEventCell.configure(viewModel: smallEvent, theme: theme)
+            cell = smallEventCell
+        }
+        
+        cell.layoutMargins = layout.itemLayoutMargins
+        estimate.height = cell.sizeThatFits(CGSize(width: columnWidth, height: UIView.noIntrinsicMetric), apply: false).height
+        estimate.precalculated = true
+        
+        return estimate
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        
+        guard let articleAsLivingDocViewModel = delegate?.articleAsLivingDocViewModel else {
+            return
+        }
+
+        (cell as? ArticleAsLivingDocLargeEventCollectionViewCell)?.delegate = self
+        (cell as? ArticleAsLivingDocLargeEventCollectionViewCell)?.articleDelegate = self
+        (cell as? ArticleAsLivingDocSmallEventCollectionViewCell)?.delegate = self
+        
+        let numSections = dataSource.numberOfSections(in: collectionView)
+        let numEvents = dataSource.collectionView(collectionView, numberOfItemsInSection: indexPath.section)
+        
+        if indexPath.section == numSections - 1 &&
+            indexPath.item == numEvents - 1 {
+            guard let nextRvStartId = articleAsLivingDocViewModel.nextRvStartId,
+                  nextRvStartId != 0 else {
+                return
+            }
+            
+            delegate?.fetchNextPage(nextRvStartId: nextRvStartId)
+        }
+    }
+    
+    @objc func collectionView(_ collectionView: UICollectionView, shouldHighlightItemAt indexPath: IndexPath) -> Bool {
+        return false
+    }
+    
+    @objc func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        return false
+    }
+
+    override func collectionViewFooterButtonWasPressed(_ collectionViewFooter: CollectionViewFooter) {
+        tappedViewFullHistoryButton()
+    }
+}
+
+// MARK:- ArticleAsLivingDocHorizontallyScrollingCellDelegate
+@available(iOS 13.0, *)
+extension ArticleAsLivingDocViewController: ArticleAsLivingDocHorizontallyScrollingCellDelegate {
+    func tappedLink(_ url: URL, cell: ArticleAsLivingDocHorizontallyScrollingCell?, sourceView: UIView, sourceRect: CGRect?) {
+        if url.absoluteString.removingPercentEncoding?.contains("/User:") == true {
+            // User talk page, should open it in a modal
```

Comment:
@mcleinman Actually we support user talk pages natively via `TalkPageContainerViewController`. Unless we specifically wanted to go to their user page and not their user talk page, we should go to the native flavor of their talk page instead.

Reply:
I assumed user page, but it seems like elsewhere in the app we got to the user talk page - I'll update this to match tomorrow.

Reply:
@mcleinman actually I think pushing to a user page is fine, I think just the comment mentioning talk threw me off. If you want to push to the user page (which I'm pretty sure is what @carolynlimadeo intended), this is the right way. If you want to go to the user _talk_ page, you should go to the native view.

Reply:
Oooh, I see now. Updated comment.